Update support info (#3400)
elastic/cloud-on-k8s: docs/supported-versions.asciidoc

* link:https://kubernetes.io/docs/tasks/tools/install-kubectl/[kubectl] 1.11+
* Kubernetes 1.12+ or OpenShift 3.11+
* Google Kubernetes Engine (GKE), Azure Kubernetes Service (AKS), and Amazon Elastic Kubernetes Service (EKS)
* Elastic Stack: 6.8+, 7.1+

> ECK should work with all conformant installers as listed in the spreadsheet link:https://github.com/cncf/k8s-conformance/blob/master/faq.md#what-is-a-distribution-hosted-platform-and-an-installer[here]. Distributions include source patches and so may not work as-is with ECK.

Reviewer: ```suggestion
ECK should work with all conformant installers as listed in these link:https://github.com/cncf/k8s-conformance/blob/master/faq.md#what-is-a-distribution-hosted-platform-and-an-installer[FAQs]. Distributions include source patches and so may not work as-is with ECK.
```
It's recommended to spell the links where possible to avoid the "here" link.
